### PR TITLE
Allow `authorise-without-apparmor` when `aa_getpeercon()` returns `ENOPROTOOPT`

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Fixes: #167

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
Confirmed working on Debian/buster

File: src/frame_main.cpp [hunk 1]
```diff
@@ -59,6 +59,7 @@ int main(int argc, char const* argv[])
             ConfigurationOption{[&](auto& option) { background_client.set_crash_background_colour(option);},
                                "diagnostic-background", "Colour of diagnostic screen background RGB", "0x380c24"},
             ConfigurationOption{[&](auto& option) { background_client.set_crash_text_colour(option);},
+
```
Comment (Saviq) [inline src/frame_main.cpp:62]:
```suggestion
```

File: src/snap_name_of.cpp [hunk 2]
```diff
@@ -41,7 +41,9 @@ auto snap_name_of(miral::Application const& app, bool fallback_without_apparmor,
     {
         mir::log_info("aa_getpeercon() failed for process %d: %s", miral::pid_of(app), strerror(errno));
 
-        if ((errno == EINVAL) && fallback_without_apparmor) // EINVAL is what is returned when AppArmor isn't setup
+        // EINVAL is what is returned when AppArmor isn't setup
+        // ENOPROTOOPT is what is returned when AppArmor doesn't have some Ubuntu patches (yet)
+        if (((errno == EINVAL)|| errno == ENOPROTOOPT) && fallback_without_apparmor)
```
Comment (Saviq) [inline src/snap_name_of.cpp:46]:
```suggestion
        if (((errno == EINVAL) || errno == ENOPROTOOPT) && fallback_without_apparmor)
```

**Review by Saviq — CHANGES_REQUESTED** [review]

**Review by Saviq — APPROVED** [review]
Yups.

**Pull request merged**